pkg/prometheus_ingester/prometheus_ingester.go

```diff
@@ -75,9 +76,48 @@ type queryOptions struct {
 	ResultAsQuantity *bool
 }
 
+type PrometheusIngesterHttpHeader struct {
+	Name            string
+	ValueFromString string `yaml:"valueFromString"`
+	ValueFromEnv    string `yaml:"valueFromEnv"`
+	Value           string `yaml:"-"`
+}
+
+func (h *PrometheusIngesterHttpHeader) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type PrometheusIngesterHttpHeaderCustom PrometheusIngesterHttpHeader
+
+	out := PrometheusIngesterHttpHeaderCustom{}
+
+	if err := unmarshal(&out); err != nil {
+		return err
+	}
+
+	*h = PrometheusIngesterHttpHeader(out)
+
+	if (h.ValueFromEnv == "") == (h.ValueFromString == "") {
```

Comment:
It should be possible to an empty value for a header.

Empty Name is a bigger problem, I'd imagine.

Reply:
fixed d48902d4ad77ea48e830ce5ce24c2a6aa5fce662